crates/primitives/src/chain/spec.rs

```diff
@@ -440,27 +439,19 @@ impl From<Vec<(Hardfork, BaseFeeParams)>> for ForkBaseFeeParams {
     }
 }
 
-/// BaseFeeParams contains the config parameters that control block base fee computation
-#[derive(Serialize, Deserialize, Debug, Copy, Clone, PartialEq, Eq)]
-pub struct BaseFeeParams {
-    /// The base_fee_max_change_denominator from EIP-1559
-    pub max_change_denominator: u64,
-    /// The elasticity multiplier from EIP-1559
-    pub elasticity_multiplier: u64,
+#[cfg(feature = "optimism")]
+pub trait OptimismBaseFeeParams {
+    fn optimism_sepolia() -> BaseFeeParams;
+    fn optimism_sepolia_canyon() -> BaseFeeParams;
+    fn optimism() -> BaseFeeParams;
+    fn optimism_canyon() -> BaseFeeParams;
 }
 
-impl BaseFeeParams {
-    /// Get the base fee parameters for Ethereum mainnet
-    pub const fn ethereum() -> BaseFeeParams {
-        BaseFeeParams {
-            max_change_denominator: EIP1559_DEFAULT_BASE_FEE_MAX_CHANGE_DENOMINATOR,
-            elasticity_multiplier: EIP1559_DEFAULT_ELASTICITY_MULTIPLIER,
-        }
-    }
-
+#[cfg(feature = "optimism")]
+impl OptimismBaseFeeParams for BaseFeeParams {
```

Comment:
we only use this inside this file, so I'd rather have constants for this:

`const OP_SEPOLIA_BASEFEE_PRARAMs = ...`